Add TomEE 10.1.1 flavors to dependencies

kiril-keranov opened the pull request:
<!-- Thanks for contributing. To speed up the process of reviewing your pull
request please provide us with the following information: -->

## Summary
Adding latest `Apache TomEE 10.1.1` [distributions](https://archive.apache.org/dist/tomee/tomee-10.1.1) to the `buildpack.toml` dependencies list in order to be able to use them with this tomee CNB. The four flavors - `microprofile`, `webprofile`, `plus` and `plume` analogically to the older major versions supported.

## Use Cases
Enables using latest `Apache TomEE 10.1.1` with tomee CNB builds.

## Checklist
<!-- Please confirm the following -->
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

kiril-keranov commented:
@loewenstein-sap @dmikusa I've tried the CNB with the added TomEE 10.1.1 flavors and was able to build applications using the 4 TomEE flavors. Seems adding the dependencies info to the `buildpack.toml` is sufficient in order to use 10.1.1. Please ping me if anything else needs to be added or modified which I am currently unaware of, thanks

dmikusa commented:
@kiril-keranov - Thanks for the PR!

First, when we add/remove dependencies, we also have to update the [pipeline-descriptor.yml](https://github.com/paketo-buildpacks/apache-tomee/blob/main/.github/pipeline-descriptor.yml) file as well. This file is what triggers the tooling to create CI pipelines to watch and submit updates for this branch.

Second, I see that we have 7, 8, 9 and we're now adding 10. It looks like 7, 8, and 9 have all reached their end of life, 9 having gone EOL Dec of 2024. https://tomee.apache.org/download-discontinued.html

It seems like we should probably do some clean up and minimally remove 7 & 8, but possibly remove 9 as well. This is typically the policy that we follow with Paketo, removing things once they go EOL upstream.

What are your thoughts on removing those?

kiril-keranov commented:
@dmikusa Leaving only version 10 and removing the older versions suits well for my use case as our stuff is based on TomEE 10.1.x. Version 8 is not supported since almost 2 years but I wonder if someone that hasn't migrated to `jakarta.*` and still uses `javax.servlet` for example might appear. I would probably leave still version 9 flavors as suggested along with 10 although its support is also ended since almost a year. I have no firm opinion on this.

kiril-keranov commented:
> OK, that was my thought as well. Drop 7 & 8, but keep 9 and 10. We can drop 9 later, but we can give folks a little more runway before doing that.
> 
> Would you mind making the following changes in your PR?
> 
> 1. Remove 7 & 8 metadata from buildpack.toml.
> 2. Update pipeline-descriptor.yml. Remove 7 & 8 and add 10.
> 
> That should be sufficient. You can technically run the tool that updates our pipelines locally, but you don't have to. The system will run that automatically after we merge your PR.
> 
> Thanks!

1. Removed 7 and 8 metadata from buildpack.toml
2. Updated `pipeline-descriptor.yml` with added info from version 10 and removed 7 and 8.
I noticed the latest 9 version that is currently available is 9.1.3 and the currently described in buildpack.toml is 9.0.0. Seems this is because of the corresponding regex in `pipeline-descriptor.yml` for version 9 - `"9\\.[\\d]+\\.[\\d]+?[-+]?[a-zA-Z0-9]+"` which seems to match for example versions with identifier like 9.0.0-M1 but not regular versions. I guess it's a bit late but maybe change it also?

kiril-keranov commented:
> Yes, can you update that as well. Thanks!
Updated
